pkg/builder/controller.go

```diff
@@ -63,32 +63,62 @@ func (blder *Builder) ForType(apiType runtime.Object) *Builder {
 	return blder.For(apiType)
 }
 
+// ForInput represents the information set by For method.
+type ForInput struct {
```

Comment:
Can we get away without exporting this type?
Likewise for `OwnsInput` and `WatchesInput`?

Reply:
If we want to allow custom implementations of options (which is nice), this has to be public

Reply:
Ah, so that others can implement `ForOption.ApplyToFor` and company. I see.